server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -178,10 +178,6 @@ public void testSortAndPaginateWithInProgress() throws Exception {
         awaitNumberOfSnapshotsInProgress(inProgressCount);
 
         final String[] repos = { repoName };
-        assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.START_TIME);
```

Comment:
`assertStablePagination` fails for in-progress snapshots, because we get a different amount of `successfulShards` for different calls, so getting all the data and getting it page by page doesn't yield the same result. I'm not sure if there's a way pinpoint the amount of successfully migrated shards. Currently, we only block the data nodes

Reply:
The issue is that some of the shards are unchanged so don't need to write any data files and therefore we don't block them from succeeding. I think we can just wait for them all to succeed:

```diff
diff --git a/server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java b/server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java
index 03fc2ad0529..d5a832f0079 100644
--- a/server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java
@@ -14,6 +14,7 @@ import org.elasticsearch.action.admin.cluster.snapshots.create.CreateSnapshotRes
 import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsRequest;
 import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsRequestBuilder;
 import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse;
+import org.elasticsearch.cluster.SnapshotsInProgress;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.search.sort.SortOrder;
 import org.elasticsearch.threadpool.ThreadPool;
@@ -176,8 +177,22 @@ public class GetSnapshotsIT extends AbstractSnapshotIntegTestCase {
             inProgressSnapshots.add(startFullSnapshot(repoName, snapshotName));
         }
         awaitNumberOfSnapshotsInProgress(inProgressCount);
+        awaitClusterState(
+            state -> state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
+                .entries()
+                .stream()
+                .flatMap(s -> s.shards().stream())
+                .allMatch(
+                    e -> e.getKey().getIndexName().equals("test-index-1") == false
+                        || e.getValue().state() == SnapshotsInProgress.ShardState.SUCCESS
+                )
+        );

         final String[] repos = { repoName };
+        assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.START_TIME);
+        assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.NAME);
+        assertStablePagination(repos, allSnapshotNames, GetSnapshotsRequest.SortBy.INDICES);
+
         assertThat(
             clusterAdmin().prepareGetSnapshots(matchAllPattern())
                 .setSnapshots(GetSnapshotsRequest.CURRENT_SNAPSHOT, "-snap*")
```

Reply:
Awesome!